packages/arch3-core/src/signingarchwayclient.ts

```diff
@@ -321,8 +321,19 @@ export class SigningArchwayClient extends SigningCosmWasmClient implements IArch
       }
     });
     const response = await this.assertSignAndBroadcast(senderAddress, [message], fee, memo);
-    const rewardsAttr = logs.findAttribute(response.parsedLogs, 'archway.rewards.v1.RewardsWithdrawEvent', 'rewards');
-    const rewards = JSON.parse(rewardsAttr.value) as Coin[];
+
+    let rewards: Coin[];
+
+    try {
+      const rewardsAttr = logs.findAttribute(response.parsedLogs, 'archway.rewards.v1.RewardsWithdrawEvent', 'rewards');
+      rewards = JSON.parse(rewardsAttr.value) as Coin[];
+    } catch (error) {
+      if ((error as Error)?.message?.includes("Could not find attribute 'rewards' in first event of type 'archway.rewards.v1.RewardsWithdrawEvent' in first log")) {
+        rewards = [];
+      }
+      else { throw error; }
+    }
```

Comment:
Here is a suggestion:

```suggestion
    const firstLogs = response.parsedLogs.find(() => true);
    const rewardsAttr = firstLogs?.events
      .find(event => event.type === 'archway.rewards.v1.RewardsWithdrawEvent')
      ?.attributes.find(attr => attr.key === 'rewards')
      ?.value;
    const rewards: Coin[] = rewardsAttr ? JSON.parse(rewardsAttr) as Coin[] : [];
```

The problem with the try/catch in this specific block is that, as you noted, it can obfuscate other real issues. However, looking at the implementation of `logs.findAttribute`, you'll see that the filter/find logic doesn't throw an error - this is actually forced by the library method.

So instead of throwing/catching/re-throwing, the code will look easier to read if we filter the log attributes directly inside the `withdrawContractRewards` method.

Reply:
Added, thanks!